### ISSUE TITLE
How do we handle different versions of NAACCR?

### ISSUE THREAD

**Issue by mgurley** [OHDSI/OncologyWG#34]
We need to calculate the delta between different versions of NAACCR.  The SEER API should be able to be used at least on the NAACCR item level and non-schema specific NAACCR item code level to give us the ability to compare different versions of NAACCR.  If NAACCR item numbers and NAACCR item codes are stable over time then the representation of NAACCR version within OMOP will be unnecessary.

**Comment by rtmill** [comment]
A good place to start: http://datadictionary.naaccr.org/?c=16

At a glance, it looks like the revisions typically add to or remove from the sets of codes rather than change the meaning of them.

**Comment by cgreich** [comment]
And what happens if they remove?

**Comment by rtmill** [comment]
@cgreich 
A couple examples:

Item #1760 Vital Status
v16:
```
0	Dead (CoC)
1	Alive
4	Dead (SEER)
```
v18:
```
0	Dead
1	Alive
```

Item #776 EOD Mets
v16
```
00	None
70	Mets present
88	Not applicable - e.g., Leukemia
99	Unknown
```

v18:
```
00	None
        No distant metastasis
        Unknown if distant metastasis
88	Not applicable: Information not collected for this schema
        Use for these sites only: HemeRetic; 
        Ill Defined Other (includes unknown primary site); 
        Kaposi Sarcoma; 
        Lymphoma; 
        Lymphoma-CLL/SLL; 
        Myeloma Plasma Cell Disorder
99	Death certificate only (DCO)
Note: This data item was included in Standards Volume II, Version 16; however, it was not implemented until 2018.
```

**Comment by cgreich** [comment]
Urgh. I am getting more and more wary about this NAACCR vocabulary. This is ugly as sin. "Dead (SEER)"? Really? DCO? 

However, in this case we can make it work. The Seer death will be remapped to the generic death, and the missing 70 Mets present will just go away. Which is funny. Looks like in V18 you cannot have metastases unless you died and there is a death certificate. Good for the patients, alas I don't buy it.

**Comment by mgurley** [comment]
At the 3/12/2019 Oncology Workgroup meeting, we decided that we will only ingest NAACCR version 18 and subsequent versions.  Each new NAACCR version will have a delta calculated and NAACCR Items and NAACCR Item codes will be deprecated going forward.

@AEW0330 rasidd that this could be a problem for groups that have NAACCR data in prior versions because of legacy systems not being migrated to the latest NAACCR version.  We decided that if the community raises this as a frequent use case, the issue of prior versions will need to be revisited.
